crates/sui-faucet/src/local_faucet.rs

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fmt;
+use std::sync::Arc;
+
+use anyhow::bail;
+use tokio::sync::Mutex;
+use tokio::time::Duration;
+use tracing::info;
+
+use crate::FaucetConfig;
+use crate::FaucetError;
+use sui_sdk::{
+    rpc_types::{SuiTransactionBlockResponse, SuiTransactionBlockResponseOptions},
+    types::quorum_driver_types::ExecuteTransactionRequestType,
+};
+
+use crate::CoinInfo;
+use shared_crypto::intent::Intent;
+use sui_keys::keystore::AccountKeystore;
+use sui_sdk::rpc_types::SuiTransactionBlockEffectsAPI;
+use sui_sdk::types::programmable_transaction_builder::ProgrammableTransactionBuilder;
+use sui_sdk::types::{
+    base_types::{ObjectID, SuiAddress},
+    gas_coin::GasCoin,
+    transaction::{Transaction, TransactionData},
+};
+use sui_sdk::wallet_context::WalletContext;
+
+const GAS_BUDGET: u64 = 10000000;
```

Comment:
nit: (please do in a followup if needed)
```suggestion
const GAS_BUDGET: u64 = 10_000_000;
```

Reply:
Fixing with the above.